server/SCHEMAS/mongodb.js

```diff
@@ -38,104 +43,270 @@ const QandASchema = new Schema({
+      photosArr.push(url)
+      newAnswer.answer_photos = photosArr
+      answers.push(newAnswer)
+      question.answers = answers
+
+      await QandA.findByIdAndUpdate(id, question).lean()
     }
-  })
+  }
+  catch (err) { return err }
+}
 
-  if (!isPresent) {
-    // add the image url to the answer photos array
-    photosArr.push(url)
-    newAnswer.answer_photos = photosArr
+// increment a helpful question
+let helpfulQuestion = async (question_id) => {
+  try {
+    let query = { question_id }
+    let question = await QandA.findOne(query).lean()
+    let id = question._id
+    let question_helpfulness = question.question_helpfulness + 1
 
-    // add the updated answer to the answers array
-    answers.push(newAnswer)
+    await QandA.findByIdAndUpdate(id, { question_helpfulness }).lean()
+  }
+  catch (err) { return err }
+}
 
-    // update the answers for the question
-    question.answers = answers
+// increment a helpful answer
+let helpfulAnswer = async (answer_id) => {
+  try {
+    // query the db for the question and needed info
+    let question = await QandA.findOne({
+      'answers': { $elemMatch: { 'answer_id': answer_id } }
+    }).lean()
+    let id = question._id
+    let answers = question.answers
+
+    answers.forEach((answer, index) => {
+      if (parseInt(answer_id) === answer.answer_id) {
+        console.log('before', answer.answer_helpfulness)
+        answer.answer_helpfulness++
+      }
+    })
 
-    await QandA.findByIdAndUpdate(id, question)
+    await QandA.findByIdAndUpdate(id, { answers }).lean()
   }
+  catch (err) { return err }
+}
+
+// report a question
+let reportedQuestion = async (question_id) => {
+  try {
+        // query the db for the question and needed info
+        let query = { question_id }
+        let question = await QandA.findOne(query).lean()
+        let id = question._id
+        let question_reported = true /* !question.question_reported */
 
+        await QandA.findByIdAndUpdate(id, { question_reported }).lean()
+  }
+  catch (err) { return err }
 }
 
-// Compile model from schema
-const QandA = mongoose.model("QandAs", QandASchema)
-const Answer = mongoose.model("Answers", AnswerSchema)
+// report an answer
+let reportedAnswer = async (answer_id) => {
+  try {
+    // query the db for the question and needed info
+    let question = await QandA.findOne({
+      'answers': {
+        $elemMatch: {
+          'answer_id': answer_id
+        }
+      }
+    }).lean()
+    let id = question._id
+    let answers = question.answers
 
+    answers.forEach((answer, index) => {
+      if (parseInt(answer_id) === answer.answer_id) {
+        answer.answer_reported = true /* !question.question_reported */
+      }
+    })
+
+    await QandA.findByIdAndUpdate(id, { answers }).lean()
+  }
+  catch (err) { return err }
+}
 
-module.exports = { QandA, Answer, insert, answerInsert, answerPhotoInsert }
\ No newline at end of file
+module.exports = {
+  QandA,
+  Answer,
+  Count,
```

Comment:
nit: Variable names could be a bit clearer here, especially for 'count' and 'Count', but not super important

Reply:
Updated for clarity, thanks :)